### PR TITLE
ANN: check `#[deprecated]` attr syntax: E0538, E0551, E0541

### PR DESCRIPTION
**Pull request by kuksag**
changelog:
* Add support for E0538, multiple items used in `#[deprecated]` attribute, e.g.:
```rust
#[deprecated(
    since = "a",
    since = "b", // error
    note = "c"
)]
fn f1() { }
```
Error code reference: https://doc.rust-lang.org/error_codes/E0538.html
Compiler reference: https://github.com/rust-lang/rust/blob/904dd2c3987028f0270db306b9964bc465689de8/compiler/rustc_attr/src/builtin.rs#L834

* Add support for E0541, unknown item used inside `#[deprecated]`:
```rust
#[deprecated(
    since="1.0.0",
    // error: unknown meta item
    reason="Example invalid meta item. Should be 'note'")
]
fn deprecated_function() {}
```
Error code reference: https://doc.rust-lang.org/error_codes/E0541.html
Compiler reference: https://github.com/rust-lang/rust/blob/904dd2c3987028f0270db306b9964bc465689de8/compiler/rustc_attr/src/builtin.rs#L891

* Add support for E0551, invalid item was used (key-value param expected):
```rust
#[deprecated(note)] // error!
fn i_am_deprecated() {}
```
Error code reference: https://doc.rust-lang.org/error_codes/E0551.html
Compiler reference: 
https://github.com/rust-lang/rust/blob/904dd2c3987028f0270db306b9964bc465689de8/compiler/rustc_attr/src/builtin.rs#L852

* Add tests for quick-fixes

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/annotator/RsAttrErrorAnnotator.kt [hunk 1]
```diff
@@ -22,13 +22,42 @@ class RsAttrErrorAnnotator : AnnotatorBase() {
...
+    val usedArgs = mutableSetOf<String>()
+    for (arg in element.metaItemArgsList) {
+        if (arg !is RsMetaItem) continue
+        val argName = arg.name ?: continue
+        if (arg.litExpr == null) {
+            RsDiagnostic.IncorrectItemInDeprecatedAttr(arg).addToHolder(holder)
+        }
+        when (argName) {
+            // "suggestion" argument is used only in compiler development
+            "since", "note", "suggestion" -> {
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/annotator/RsAttrErrorAnnotator.kt:47]: I suppose this check should be moved inside `"since", "note", "suggestion" -> {` branch since rustc shows only `error[E0551]: incorrect meta item` in this case: `#[deprecated(foobarbranch)]`.
Also, I think this error should not be shown in the case of `#[deprecated(since = )]` since it's a kind o f syntax error, rustc shows "expected unsuffixed literal or identifier, found `<eof>`" in this case

```suggestion
        when (argName) {
            if (arg.eq == null) {
                RsDiagnostic.IncorrectItemInDeprecatedAttr(arg).addToHolder(holder)
            }
            // "suggestion" argument is used only in compiler development
            "since", "note", "suggestion" -> {
```

**Review by Undin — COMMENTED** [review]
@kuksag let's avoid merging master branch into feature branch. Use rebase, please

File: src/test/kotlin/org/rust/ide/annotator/RsAttrErrorAnnotatorTest.kt [hunk 2]
```diff
@@ -144,4 +160,82 @@ class RsAttrErrorAnnotatorTest : RsAnnotatorTestBase(RsAttrErrorAnnotator::class
...
+        #[deprecated(
+            since = "a",
+            <error descr="Multiple 'since' items [E0538]">since = "b"/*caret*/</error>,
+            note = "c"
+        )]
+        fn f1() { }
+    """, """
+        #[deprecated(
+            since = "a",
+        ,
```
Comment (vlad20012) [inline src/test/kotlin/org/rust/ide/annotator/RsAttrErrorAnnotatorTest.kt:214]: The quick fix should also remove the comma. I think it's better just remove the quick-fix, merge the PR without it, and maybe make a separate PR with the adjusted quick-fix
